pkg/operator/telemetry/telemetry.go

```diff
@@ -0,0 +1,242 @@
+	// instanceIDKey is the key used to store the instance ID in the ConfigMap
+	instanceIDKey = "instance-id"
+)
+
+// Service handles telemetry operations for the operator
+type Service struct {
+	client        client.Client
+	versionClient updates.VersionClient
+	namespace     string
+}
+
+// LeaderTelemetryRunnable runs telemetry checks only when this instance is the leader
+type LeaderTelemetryRunnable struct {
+	TelemetryService *Service
+}
+
+// Start starts the telemetry runner
+func (t *LeaderTelemetryRunnable) Start(ctx context.Context) error {
+	ctxLogger := log.FromContext(ctx)
+	ctxLogger.Info("Leader elected, starting telemetry worker")
+
+	// Start telemetry worker in a goroutine with the leader context
+	// When leadership is lost, ctx will be cancelled and telemetry will stop
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				ctxLogger.Error(fmt.Errorf("telemetry worker panic: %v", r), "Telemetry worker panicked")
+			}
+		}()
+		t.TelemetryService.StartTelemetryWorker(ctx)
+	}()
+
+	// Wait for context cancellation (leadership lost or shutdown)
+	<-ctx.Done()
+	ctxLogger.Info("Leadership lost, telemetry worker stopped")
+	return nil
+}
+
+// NeedsLeaderElection indicates whether this runnable needs leader election
+func (*LeaderTelemetryRunnable) NeedsLeaderElection() bool {
+	// This runnable should only run when this instance is the leader
+	return true
+}
+
+// telemetryData represents the structure of telemetry data stored in ConfigMap
+type telemetryData struct {
+	InstanceID      string    `json:"instance_id"`
+	LastUpdateCheck time.Time `json:"last_update_check"`
+	LatestVersion   string    `json:"latest_version"`
+}
+
+// NewService creates a new Service instance
+func NewService(k8sClient client.Client, namespace string) *Service {
+	if namespace == "" {
+		namespace = configMapNamespace
+	}
+	return &Service{
+		client:        k8sClient,
+		versionClient: updates.NewVersionClientWithSuffix("operator"),
+		namespace:     namespace,
+	}
+}
+
+// CheckForUpdates checks for updates and sends telemetry data
+func (s *Service) CheckForUpdates(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	// Get or create telemetry data
+	data, err := s.getTelemetryData(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get telemetry data: %w", err)
+	}
+
+	// Check if we need to make an API request based on last update time
+	if time.Since(data.LastUpdateCheck) < updateInterval {
+		// Too soon, skip the check
+		logger.V(1).Info("Skipping update check, too soon since last check",
+			"lastCheck", data.LastUpdateCheck,
+			"interval", updateInterval)
+		return nil
+	}
+
+	logger.Info("Checking for updates...")
+
+	// Get the latest version from the API
+	currentVersion := versions.GetVersionInfo().Version
+	latestVersion, err := s.versionClient.GetLatestVersion(data.InstanceID, currentVersion)
+	if err != nil {
+		logger.Error(err, "Failed to check for updates")
```

Comment:
We shouldn't need this logging line as the calling functions of `CheckForUpdates()` log the error anyways

Reply:
removed it 👍 Thanks!